MapboxDirections/Match/MBTracepoint.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+@objc(MBTracepoint)
+public class Tracepoint: Waypoint {
+    
+    /**
+     Number of probable alternative matchings for this trace point. A value of zero indicates that this point was matched unambiguously.
+     */
+    @objc open var alternateCount: Int
+    
+    /**
+     Index of the waypoint inside the matched route.
+     */
+    open var waypointIndex: Int?
+    
+    /**
+      Index to the match object in matchings the sub-trace was matched to.
+     */
+    @objc open var matchingIndex: Int
+    
+    init(coordinate: CLLocationCoordinate2D, alternateCount: Int, waypointIndex: Int?, matchingIndex: Int, name: String?) {
+        self.alternateCount = alternateCount
+        self.waypointIndex = waypointIndex
+        self.matchingIndex = matchingIndex
+        super.init(coordinate: coordinate)
+        self.name = name
```

Comment:
Call the initializer that takes a CLLocation instead; that’ll set `name` automatically. Along the way, this initializer could take a CLLocation too.

Reply:
I don't have access to a CLLocation at this point though. You think it's worth it to create one?

Reply:
After the call to `CLLocation(coordinate:)`, this object has an initialized `location` property of type CLLocation.